hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -513,6 +525,31 @@ public Map<String, SubClusterPolicyConfiguration> invoke(
     return cacheRequest;
   }
 
+  private Map<ApplicationId, SubClusterId> buildApplicationSubClusterMap(
+      GetApplicationHomeSubClusterResponse response) {
+    ApplicationHomeSubCluster applicationHomeSubCluster = response.getApplicationHomeSubCluster();
+    SubClusterId subClusterId = applicationHomeSubCluster.getHomeSubCluster();
+    ApplicationId applicationId = applicationHomeSubCluster.getApplicationId();
+    Map<ApplicationId, SubClusterId> appSubCluster = new HashMap<>();
+    appSubCluster.put(applicationId, subClusterId);
+    return appSubCluster;
+  }
+
+  private Object buildGetApplicationHomeSubClusterRequest(String appId) {
+    final String cacheKey = buildCacheKey(getClass().getSimpleName(),
+        GET_APPLICATION_HOME_SUBCLUSTER_CACHEID, appId);
+    CacheRequest<String, Map<ApplicationId, SubClusterId>> cacheRequest =
+        new CacheRequest<>(cacheKey,
+            input -> {
+              ApplicationId applicationId = ApplicationId.fromString(appId);
+              GetApplicationHomeSubClusterResponse response =
+                  stateStore.getApplicationHomeSubCluster(
+                  GetApplicationHomeSubClusterRequest.newInstance(applicationId));
```

Comment:
The indentation is not correct.

Reply:
Let's create a method that takes string and does the appId transformation inside.

Reply:
Thanks for your suggestion, I will fix it.